.github/actions/install_node_dependencies/action.yml

```diff
@@ -0,0 +1,22 @@
+name: Setup Node.js and Install Dependencies
+description: Sets up a specific Node.js version, caches Node modules, and installs Node dependencies.
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-node@v4.0.0
+      with:
+        node-version: 16.13.2
+
+    - id: npm-cache-dir
+      run: echo "dir=$(npm config get cache)" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - uses: actions/cache@v4
+      with:
+        path: "${{ steps.npm-cache-dir.outputs.dir }}"
+        key: "${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}"
+        restore-keys: "${{ runner.os }}-node-"
+
+    - run: npm ci
```

Comment:
[Nit] To make this reusable may be we can have path to package.json as input which defaults to current directory